Lib/dbm/__init__.py

```diff
@@ -49,6 +49,13 @@ class error(Exception):
 except ImportError:
     ndbm = None
 
+try:
+    import sqlite3
```

Comment:
People using `dbm` are only likely to use a single backend. It is better to delay the import of the specific implementation until it is going to be used.  I'd just always add dbm.sqlite3 to _names above and adjust the loop checking for ImportError within `open()` below if neecessary at all.  An ImportError coming from the attempt to load `dbm.sqlite3` when someone needs or asks for it is sufficient.

Reply:
Hm, yeah. That's also similar to how the other backends are handled; for example, `dbm.gnu` is always present in `_names`, even though not all systems have the `_gdbm` module.

Reply:
This reminds me we'd better have some more tests for `.whichdb()`, to make sure SQLite databases are autodetected properly.

Reply:
Addressed in b655473dbd199006945c2aab8e754684ba7f1fec and 95eba9348c8939cc714881b2ac08648aaf0bfad6.